### PR TITLE
Create find_duplicates.py

### PR DESCRIPTION
**Pull request by AlienKevin**
Finds all duplicated lines within the CSV files. Aims to close [issue 159](https://github.com/rime/rime-cantonese/issues/159) in [rime-cantonese](https://github.com/rime/rime-cantonese/issues/159)

### CODE REVIEW

**Review by ayaka14732 — CHANGES_REQUESTED** [review]
Thank you! You also need to modify the YAML files in `.github/workflows`.

**Comment by AlienKevin** [comment]
@ayaka14732 I added `find_duplicates` to the workflows. Though I'm not sure about how the workflow actually runs. Currently, I changed the script to output error messages to stdout, like `validate.py`. There are 859 duplicated lines right now so my console is overflown when I run the script. You might need to redirect the messages to some log file for inspection.

**Comment by ayaka14732** [comment]
@AlienKevin If there exists duplicate entries in the dictionary, the script should terminate with a non-zero exit code, so that we will get a notification saying that the workflow run has failed.
@laubonghaudoi We should remove the duplicate entries first, then merge the changed dictionaries back to this branch.

**Comment by laubonghaudoi** [comment]
@AlienKevin @ayaka14732 啱先刪咗重複詞條，你哋可以繼續嘞 https://github.com/CanCLID/rime-cantonese-upstream/commit/a80ee507406660331e6e7f9d0c79b0ef4db36ed5

**Comment by ayaka14732** [comment]
Why has the check passed? I thought https://github.com/CanCLID/rime-cantonese-upstream/commit/a80ee507406660331e6e7f9d0c79b0ef4db36ed5 has not been merged to this branch yet.

**Comment by laubonghaudoi** [comment]
係唔係 @AlienKevin 已經將最新版 pull 咗落呢個 PR

**Comment by AlienKevin** [comment]
呢個都幾奇怪，我冇pull過。會唔會checkout個action係自動checkout main分支嘅最新版本？

**Review by ayaka14732 — APPROVED** [review]
個 script 又好似冇咩問題

**Comment by ayaka14732** [comment]
> @ayaka14732 綾香你會唔會偏好用 `defaultdict`？

Oh, I didn't noticed. Should use `defaultdict` instead.

**Comment by AlienKevin** [comment]
Didn't know `defaultdict` exists. Seems like a neat feature. Updated to use `defaultdict` instead of `dict`.

**Comment by graphemecluster** [comment]
@AlienKevin You can pass the `list` constructor directly to the `defaultdict` constructor, I think it is the more common practice

File: scripts/find_duplicates.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+到未啊,dou3 mei6 aa3	[phrase_fragment.csv:1830, word.csv:11234]
+
+This means that the line "到未啊,dou3 mei6 aa3" is duplicated
+in phrase_fragment.csv on line 1830 and in word.csv on line 11234.
+"""
+import sys
+from glob import iglob
+from collections import defaultdict
+
+line_to_locations = defaultdict(lambda: [])
```
Comment (ayaka14732) [inline scripts/find_duplicates.py:16]:
```suggestion
line_to_locations = defaultdict(list)
```

File: scripts/find_duplicates.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+A sample line in the output looks like:
+到未啊,dou3 mei6 aa3	[phrase_fragment.csv:1830, word.csv:11234]
+
+This means that the line "到未啊,dou3 mei6 aa3" is duplicated
+in phrase_fragment.csv on line 1830 and in word.csv on line 11234.
+"""
+import sys
+from glob import iglob
+from collections import defaultdict
```
Comment (ayaka14732) [inline scripts/find_duplicates.py:14]: Sort imports:

```suggestion
from collections import defaultdict
from glob import iglob
import sys
```

**Review by ayaka14732 — CHANGES_REQUESTED** [review]

**Comment by AlienKevin** [comment]
@graphemecluster Got it. @ayaka14732 Just applied your suggestions as a new commit.

**Pull request merged**